modules/dyn.js

```diff
@@ -0,0 +1,242 @@
+// dyn.js
+
+import {
+  pullComments, pullCommCounter, pushComment, pushLike, pullLikes,
+} from './socialApi.js';
+
+export default class DynGrid {
+  header = document.getElementById('Header');
+
+  footer = document.getElementById('Footer');
+
+  dynamicGrid = document.getElementById('dynamicGrid');
+
+  genCard = (data, i, unqId, imgSrc, foodName) => {
+    const id = i + 1;
+    const cardContainer = document.createElement('div');
+    const cardPic = document.createElement('div');
+    const cardPicImg = document.createElement('img');
+    cardPicImg.src = `${imgSrc}`;
+    const cardMeta = document.createElement('div');
+    const cardInfo = document.createElement('div');
+    const cardTitle = document.createElement('div');
+    cardTitle.innerHTML = `${foodName}`;
+    cardPic.classList.add('card-pic');
+    cardMeta.classList.add('card-meta');
+    cardInfo.classList.add('card-info');
+    cardTitle.classList.add('card-title');
+    const cardSocials = document.createElement('div');
+    cardSocials.classList.add('card-socials');
+    const likeBtn = document.createElement('img');
+    likeBtn.classList.add('like-btn');
+    likeBtn.src = '../src/icons/like.png';
+    likeBtn.alt = 'Like button';
+    const likeCounter = document.createElement('div');
+    likeCounter.classList.add('like-counter');
+    const showLikeQt = async (data, i, unqId) => {
+      likeCounter.innerHTML = `${await Promise.resolve(pullLikes(unqId))} likes`;
+    };
+    showLikeQt(data, i, unqId);
+    likeBtn.addEventListener('click', async () => {
+      await Promise.resolve(pushLike(unqId));
+      likeCounter.innerHTML = `${await Promise.resolve(pullLikes(unqId))} likes`;
+    });
+    const commentBtn = document.createElement('div');
+    commentBtn.id = `${id}`;
+    commentBtn.innerHTML = 'Comments';
+    const reserveBtn = document.createElement('div');
+    reserveBtn.innerHTML = 'Reservations';
+    commentBtn.classList.add('comment-btn');
+    reserveBtn.classList.add('reserve-btn');
+
+    cardContainer.className = 'card-container';
+    cardContainer.id = `${unqId}`;
+    this.dynamicGrid.className = 'dynamic-grid';
+
+    cardPic.appendChild(cardPicImg);
+    cardSocials.appendChild(likeBtn);
+    cardSocials.appendChild(likeCounter);
+    cardInfo.appendChild(cardTitle);
+    cardInfo.appendChild(cardSocials);
+    cardMeta.appendChild(cardInfo);
+    cardMeta.appendChild(commentBtn);
+    cardMeta.appendChild(reserveBtn);
+    cardContainer.appendChild(cardPic);
+    cardContainer.appendChild(cardMeta);
+    this.dynamicGrid.appendChild(cardContainer);
+
+    commentBtn.addEventListener('click', () => {
+      this.showPopup(data, i);
+    });
+  }
+
+  paintComments = (id, commentBoard, newC) => {
+    if (newC) {
+      const tmpCommBoardContent = document.getElementById('commBoardContent');
+      commentBoard.removeChild(tmpCommBoardContent);
```

Comment:
- [ ] You did a great job on the project functionalities but there is a little issue after adding a comment to any dish the comment is not visible and the counter is not updated, I need to close the comment pop-up and open it again to see my added comment and updated comment counter. Please make sure when users add any comment the pop-up is updated.
  - To fix this you can try, when the user adds a comment, calls the Involvement API again that will also include the newly added comment by the user and repopulate the comment pop-up.

### See the image of the requirement

![image](https://user-images.githubusercontent.com/83803414/163669306-8ee613ee-6bb7-4da0-be6f-e0b80e08a358.png)